freqtrade/utils.py

```diff
@@ -0,0 +1,44 @@
+import logging
+from argparse import Namespace
+from typing import Any, Dict
+
+from freqtrade.configuration import Configuration
+from freqtrade.exchange import supported_exchanges
+from freqtrade.state import RunMode
+
+
+logger = logging.getLogger(__name__)
+
+
+def setup_configuration(args: Namespace, method: RunMode) -> Dict[str, Any]:
+    """
+    Prepare the configuration for the Hyperopt module
+    :param args: Cli args from Arguments()
+    :return: Configuration
+    """
+    configuration = Configuration(args, method)
+    config = configuration.load_config()
+
+    # Ensure we do not use Exchange credentials
+    config['exchange']['key'] = ''
+    config['exchange']['secret'] = ''
+
+    return config
+
+
+def start_list_exchanges(args: Namespace) -> None:
+    """
+    Start listing known exchanges
+    :param args: Cli args from Arguments()
+    :return: None
+    """
+
+    # Initialize configuration
+    _ = setup_configuration(args, RunMode.OTHER)
```

Comment:
I don't think you need this.
`supported_exchanges()` only uses the static method from ccxt - no config or anything involved at all.

Reply:
Personally, i would remove this call, since it's not needed for this (leave the setup_configuration part in maybe, we'll need that for other things...)

It causes problem when running `freqtrade list-exchanges` without a config.json available. 
i don't, as i have ~20 i use for testing, but (intentionally) none with the default name.

I assume "list-exchanges" is one of the first thing you run when getting started (to actually decide on how to bulid a configuration), so you probably don't have a configuration yet, so this is confusing.

Apart from that - i would leave the file as `utils.py` for now (we can move it to a folder/module later, when we have more commands...

Reply:
> Personally, i would remove this call, since it's not needed for this (leave the setup_configuration part in maybe, we'll need that for other things...)

Already done it. It (the configuration) will be used in `list-pairs` to initialize exchange.

The `setup_configuration()` may be used in `start_hyperopt/backtesting` and in all new subcommands, like `list-pairs`, `plot-*`, `download`, `inspect-datadir`, etc.